simulator/src/testbench.js

```diff
@@ -860,29 +1024,33 @@ function triggerReset(reset, scope) {
  * @param {Object} data - Object containing the test data
  */
 function setUITableHeaders(testbenchData) {
-    const data = testbenchData.testData;
-    const inputCount = data.groups[0].inputs.length;
-    const outputCount = data.groups[0].outputs.length;
+    try {
+        const data = testbenchData.testData;
+        const inputCount = data.groups[0].inputs.length;
+        const outputCount = data.groups[0].outputs.length;
 
-    $('#tb-manual-table-inputs-head').attr('colspan', inputCount);
-    $('#tb-manual-table-outputs-head').attr('colspan', outputCount);
+        $('#tb-manual-table-inputs-head').attr('colspan', inputCount);
+        $('#tb-manual-table-outputs-head').attr('colspan', outputCount);
 
-    $('.testbench-runall-label').css('display', 'none');
+        $('.testbench-runall-label').css('display', 'none');
 
-    $('.tb-data#data-title').children().eq(1).text(data.title || 'Untitled');
-    $('.tb-data#data-type').children().eq(1).text(data.type === 'comb' ? 'Combinational' : 'Sequential');
+        $('.tb-data#data-title').children().eq(1).text(data.title || 'Untitled');
+        $('.tb-data#data-type').children().eq(1).text(data.type === 'comb' ? 'Combinational' : 'Sequential');
 
-    $('#tb-manual-table-labels').html('<th>LABELS</th>');
-    $('#tb-manual-table-bitwidths').html('<td>Bitwidth</td>');
+        $('#tb-manual-table-labels').html('<th>LABELS</th>');
+        $('#tb-manual-table-bitwidths').html('<td>Bitwidth</td>');
 
-    data.groups[0].inputs.concat(data.groups[0].outputs).forEach((io) => {
-        const label = `<th>${escapeHtml(io.label)}</th>`;
-        const bw = `<td>${escapeHtml(io.bitWidth.toString())}</td>`;
-        $('#tb-manual-table-labels').append(label);
-        $('#tb-manual-table-bitwidths').append(bw);
-    });
+        data.groups[0].inputs.concat(data.groups[0].outputs).forEach((io) => {
+            const label = `<th>${escapeHtml(io.label)}</th>`;
+            const bw = `<td>${escapeHtml(io.bitWidth.toString())}</td>`;
+            $('#tb-manual-table-labels').append(label);
+            $('#tb-manual-table-bitwidths').append(bw);
+        });
 
-    setUICurrentCase(testbenchData);
+        setUICurrentCase(testbenchData);
+    } catch (error) {
+        console.error("Error in setUITableHeaders:", error);
+    }
```

Comment:
I'll hold off on adding user-facing error messages in these UI functions for now. Since we're already capturing these errors in Sentry, we can monitor and address them as needed without potentially confusing the user with technical error messages. Logging to the console combined with Sentry should provide sufficient insight for debugging. If we expand user-friendly error handling across the UI in the future, we can revisit this approach.